db/dbtypes/types.go

```diff
@@ -9,10 +9,43 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/decred/dcrdata/v3/db/dbtypes/internal"
 )
 
+// TimeDef is time.Time wrapper that formats time by default as a string without
+// a timezone. The time Stringer interface formats the time into a string
+// with a timezone.
+type TimeDef struct {
+	T time.Time
+}
+
+func (t TimeDef) String() string {
+	return t.T.Format("2006-01-02 15:04:05")
+}
+
+// MarshalJSON is set as the default marshalling function for TimeDef struct.
+func (t *TimeDef) MarshalJSON() ([]byte, error) {
+	return json.Marshal(t.String())
+}
+
+// TimeAPI is a fall back dbtypes.TimeDef wrapper that allows API endpoint that
+// previously accepted timestamp instead of a formatted string time to continue
+// working.
+type TimeAPI struct {
+	S TimeDef
+}
+
+func (t TimeAPI) String() string {
+	return t.S.T.Format("2006-01-02 15:04:05")
```

Comment:
I'd suggest to return `t.S.String()` instead.

However if `TimeAPI` embedded `TimeDef` as in

```go
type TimeAPI struct {
    TimeDef
}
```

(without naming it `S`), then you would not need to implement `(t TimeAPI) String()` as it would get it from `TimeDef`.  You would still need to implement `(t *TimeAPI) MarshalJSON` of course, but it would be just `return json.Marshal(t.TimeDef.T.Unix())`.

I understand defining `TimeAPI` in this package though if it helps deal with dbtypes depending on apitypes.

Reply:
```
type TimeAPI struct {
    TimeDef
}
```
This version of the struct made my linter  to complain that I was doing unkeyed struct assignment. The keyed entry made it easier to access much of the time functionality that is accessible via the wrapper.

Reply:
I won't push my opinion further, but note that the solution is

`TimeAPI{TimeDef: t}`

Regarding accessing time functionality, you still do stuff like `t.TimeDef.T.Unix()`.  The reason I suggested this is because the definition of String is redundant.

Reply:
Let me try it. It looks easier than my current implementation.

Reply:
Just discovered that this might not work as intended because I moved `TimeAPI` struct to `apitypes` package.

Reply:
Oh, right.  Well let's just revert back and save it for another day, unless you wanted to try something else.